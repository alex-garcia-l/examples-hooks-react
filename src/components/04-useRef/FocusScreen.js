import React, { useRef } from 'react';

export const FocusScreen = () => {

  const inputRef = useRef();

  const handleClick = (item) => {

    if (item === 'nombre') {
      inputRef.current.select();
    } else {
      document.getElementById('apellido').select();
    }

  }

  return (
    <>
      <h1 className="mt-5">Focus Screen</h1>
      <hr />

      <input
        className="form-control"
        placeholder="Ingrese su nombre"
        ref={inputRef}
      />

      <input
        className="form-control mt-3"
        placeholder="Ingrese su apellido"
        id="apellido"
      />

      <button
        className="btn btn-outline-primary mt-3 me-3"
        onClick={() => handleClick('nombre')}
      >
        Seleccionar nombre
      </button>

      <button
        className="btn btn-outline-primary mt-3"
        onClick={() => handleClick('apellido')}
      >
        Seleccionar apellido
      </button>
    </>
  );
};
