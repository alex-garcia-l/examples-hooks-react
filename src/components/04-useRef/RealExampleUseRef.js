import React, { useState } from 'react';
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';

export const RealExampleUseRef = () => {

  const [visibility, setVisibility] = useState(true);

  return (
    <>
      <div className="clearfix">
        <h1 className="float-start">RealExampleUseRef</h1>
        <button
          className="btn btn-outline-primary mt-3 float-end"
          onClick={() => {
            setVisibility(!visibility);
          }}
        >Visualizar/Ocultar</button>
      </div>
      <hr />

      {visibility && <MultipleCustomHooks />}

    </>
  );
};
