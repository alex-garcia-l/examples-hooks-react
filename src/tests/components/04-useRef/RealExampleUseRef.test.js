import { shallow } from 'enzyme';
import { RealExampleUseRef } from '../../../components/04-useRef/RealExampleUseRef';

describe('Prueba para el componente <RealExampleUseRef />', () => {
  
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<RealExampleUseRef />);
  })

  test('debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
  });

  test('debe ocultar el componente MultipleCustomHooks al precionar el botÃ³n Visualizar/Ocultar', () => {
    wrapper.find('button').simulate('click');
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
  });
});
