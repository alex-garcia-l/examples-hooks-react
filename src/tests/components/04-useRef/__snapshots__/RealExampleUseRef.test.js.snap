// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba para el componente <RealExampleUseRef /> debe mostrarse correctamente 1`] = `
<Fragment>
  <div
    className="clearfix"
  >
    <h1
      className="float-start"
    >
      RealExampleUseRef
    </h1>
    <button
      className="btn btn-outline-primary mt-3 float-end"
      onClick={[Function]}
    >
      Visualizar/Ocultar
    </button>
  </div>
  <hr />
  <MultipleCustomHooks />
</Fragment>
`;

exports[`Prueba para el componente <RealExampleUseRef /> debe ocultar el componente MultipleCustomHooks al precionar el bot√≥n Visualizar/Ocultar 1`] = `
<Fragment>
  <div
    className="clearfix"
  >
    <h1
      className="float-start"
    >
      RealExampleUseRef
    </h1>
    <button
      className="btn btn-outline-primary mt-3 float-end"
      onClick={[Function]}
    >
      Visualizar/Ocultar
    </button>
  </div>
  <hr />
</Fragment>
`;
